packages/localize/src/number/forceAddGroupSeparators.js

```diff
@@ -7,9 +7,17 @@
  */
 export function forceAddGroupSeparators(formattedParts, groupSeparator) {
   let concatArray = [];
-  if (formattedParts[0].type === 'integer') {
-    const getInteger = formattedParts.splice(0, 1);
-    const numberOfDigits = getInteger[0].value.length;
+  let firstPart;
+  let integerPart;
+
+  for (let i = 0; i < formattedParts.length; i += 1) {
+    if (formattedParts[i].type === 'integer') {
+      firstPart = formattedParts.splice(0, i);
+      integerPart = formattedParts.splice(0, 1);
+    }
+  }
+  if (integerPart[0].value) {
```

Comment:
Will still throw..

Uncaught TypeError: Cannot read property '0' of undefined
    at <anonymous>:1:24

I'm curious why my suggestion doesn't work. In the old diff the same thing happens: it returns concatArray if type of the part is not 'integer'

Reply:
Ahh I know why, because in my suggestion it returns already on the first part, even though the second part might be the integer, which is the case for negative numbers. So then try:
`if (integerPart !== undefined)`

Reply:
Important to strict check for specifically undefined (rather than falsy so that you don't accidentally match with `0` 🙈 )